lib/widgets/emoji_reaction.dart

```diff
@@ -595,3 +635,446 @@ class EmojiPickerListEntry extends StatelessWidget {
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final message = PerAccountStoreWidget.of(context).messages[messageId];
+
+    final reactions = message?.reactions;
+
+    if (reactions == null || reactions.aggregated.isEmpty) {
+      return Padding(
+        padding: const EdgeInsets.only(top: 8),
+        child: BottomSheetHeaderPlainText(text: zulipLocalizations.seeWhoReactedSheetNoReactions),
+      );
+    }
+
+    return Padding(
+      padding: const EdgeInsets.only(top: 16, bottom: 4),
+      child: InsetShadowBox(start: 8, end: 8,
+        color: designVariables.bgContextMenu,
+        child: SingleChildScrollView(
+          // TODO(upstream) we want to pass excludeFromSemantics: true
+          //    to the underlying Scrollable to remove an unwanted node
+          //    in accessibility focus traversal.
+          scrollDirection: Axis.horizontal,
+          child: Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 8),
+            child: Semantics(
+              role: SemanticsRole.tabBar,
+              container: true,
+              explicitChildNodes: true,
+              label: zulipLocalizations.seeWhoReactedSheetHeaderLabel(reactions.total),
+              child: Row(
+                children: reactions.aggregated.mapIndexed((i, r) =>
+                  _ViewReactionsEmojiItem(
+                    reactionWithVotes: r,
+                    position: _emojiItemPosition(i, reactions.aggregated.length),
+                    selected: r.reactionType == reactionType && r.emojiCode == emojiCode,
+                    onRequestSelect: onRequestSelect),
+                ).toList()))))));
+  }
+}
+
+class _ViewReactionsEmojiItem extends StatelessWidget {
+  const _ViewReactionsEmojiItem({
+    required this.reactionWithVotes,
+    required this.position,
+    required this.selected,
+    required this.onRequestSelect,
+  });
+
+  final ReactionWithVotes reactionWithVotes;
+  final double position;
+  final bool selected;
+  final void Function(ReactionWithVotes) onRequestSelect;
+
+  static const double emojiSize = 24;
+
+  /// Animates the list's scroll position for this item.
+  ///
+  /// This serves two purposes when the list is longer than the viewport width:
+  /// - Ensures the item is in view
+  /// - By animating, draws attention to the fact that this is a scrollable list
+  ///   and there may be more items in view. (In particular, does this when
+  ///   any item is tapped, because each item has a different [position].)
+  void _scrollIntoView(BuildContext context) {
+    final scrollPosition = Scrollable.of(context, axis: Axis.horizontal).position;
+    final destination = lerpDouble(
+      scrollPosition.minScrollExtent,
+      scrollPosition.maxScrollExtent,
+      position);
+    if (destination == null) return; // TODO(log)
```

Comment:
This condition is impossible, right? (From the definition of lerpDouble.) So just use `!`.